communication/nats_dialog/codec_signer.go

```diff
@@ -0,0 +1,62 @@
+package nats_dialog
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/mysterium/node/communication"
+	"github.com/mysterium/node/identity"
+)
+
+func NewCodecSigner(
+	codecPacker communication.Codec,
+	signer identity.Signer,
+	verifier identity.Verifier,
+) *codecSigner {
+	return &codecSigner{
+		codecPacker: codecPacker,
+		signer:      signer,
+		verifier:    verifier,
+	}
+}
+
+type codecSigner struct {
```

Comment:
This name hides that it not only **signs** outgoing messages, but also **verifies** incoming ones.
Maybe `SignedCodec` abstracts sign+verify better?

Reply:
What about SecuredCodec ?

Reply:
Renamed to `codecSecured`